build.cake

```diff
@@ -0,0 +1,213 @@
+var NUNITLITE_TESTS = "nunitlite.tests.exe";
+var ENGINE_TESTS = "nunit.engine.tests.dll";
+var ADDIN_TESTS = "addins/tests/addin-tests.dll";
+var V2_DRIVER_TESTS = "addins/v2-tests/nunit.v2.driver.tests.dll";
+var CONSOLE_TESTS = "nunit3-console.tests.dll";
+
+///////////////////////////////////////////////////////////////////////////////
+// SETUP / TEARDOWN
+///////////////////////////////////////////////////////////////////////////////
+
+Setup(() =>
+{
+    Information("Building version {0} of Nunit.", packageVersion);
+});
+
+//////////////////////////////////////////////////////////////////////
+// TASKS
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectory(binDir);
+});
+
+
+Task("Restore-NuGet-Packages")
+    .Does(() =>
+{
+    NuGetRestore("./nunit.sln");
+});
+
+Task("Build")
+    .IsDependentOn("Restore-NuGet-Packages")
+    .Does(() =>
+{
+    if(IsRunningOnWindows())
+    {
+        // Use MSBuild
+        MSBuild("./nunit.sln", new MSBuildSettings()
+            .SetConfiguration(configuration)
+            .SetPlatformTarget(PlatformTarget.MSIL)
+//            .WithProperty("TreatWarningsAsErrors", "true")
+            .SetVerbosity(Verbosity.Minimal)
+            .SetNodeReuse(false)
+        );
+    }
+    else
+    {
+        // Use XBuild
+        XBuild("./nunit.sln", new XBuildSettings()
+            .SetConfiguration(configuration)
+            .WithTarget("AnyCPU")
+//            .WithProperty("TreatWarningsAsErrors", "true")
+            .SetVerbosity(Verbosity.Minimal)
+        );
+    }
+});
+
+Task("TestAllFrameworks")
+    .IsDependentOn("Build")
+	.Does(() =>
+	{
+		foreach(string runtime in AllFrameworks)
+		{
+			RunFrameworkTests(runtime);
+			RunNUnitLiteTests(runtime);
+		}
+	});
+
+Task("TestFramework")
+    .IsDependentOn("Build")
+	.Does(() => 
+	{ 
+		RunFrameworkTests(framework);
+	});
+
+Task("TestNUnitLite")
+    .IsDependentOn("Build")
+	.Does(() => 
+	{ 
+		RunNUnitLiteTests(framework);
+	});
+
+Task("TestEngine")
+    .IsDependentOn("Build")
+    .Does(() => 
+	{ 
+		RunTestWithConsoleRunner(ENGINE_TESTS);
+	});
+
+Task("TestAddins")
+    .IsDependentOn("Build")
+    .Does(() => 
+	{
+		RunTestWithConsoleRunner(ADDIN_TESTS); 
+	});
+
+Task("TestV2Driver")
+    .IsDependentOn("Build")
+    .Does(() => 
+	{ 
+		RunTestWithConsoleRunner(V2_DRIVER_TESTS);
+	});
+
+Task("TestConsole")
+    .IsDependentOn("Build")
+    .Does(() => 
+	{ 
+		RunTestWithConsoleRunner(CONSOLE_TESTS);
+	});
+
+//////////////////////////////////////////////////////////////////////
+// HELPER METHODS
+//////////////////////////////////////////////////////////////////////
+
+void RunExecutableTest(FilePath exePath)
+{
+	StartProcess(exePath, new ProcessSettings()
+	{
+		WorkingDirectory = binDir
+	});
+}
+
+void RunTestWithConsoleRunner(string testAssembly)
+{
+	StartProcess(NUNIT3_CONSOLE, new ProcessSettings()
+	{
```

Comment:
This is where an error arises using -mono. NUNIT3_CONSOLE and other global vars are not known to the compiler at this point when using the mono compiler.